dace/transformation/dataflow/copy_to_map.py

```diff
@@ -31,7 +36,12 @@ def can_be_applied(self, graph: SDFGState, expr_index: int, sdfg: SDFG, permissi
         if isinstance(self.b.desc(sdfg), data.View):
             if sdutil.get_view_node(graph, self.b) == self.a:
                 return False
-        if self.a.desc(sdfg).strides == self.b.desc(sdfg).strides:
+        if (not self.ignore_strides) and self.a.desc(sdfg).strides == self.b.desc(sdfg).strides:
+            return False
+        if self.a.data == self.b.data:
```

Comment:
Why are we disallowing this? I understand that alarms should light up if we are copying from/to the same container. However, any potential damage is already done if the copy is already on the SDFG. Therefore, I would expect that, in a correct SDFG, to not actually have an overlap among the subsets.

Reply:
I am not fully understand your comment about the same container.
Furthermore, I also do not understand why this test of the strides is, for me it does not make any sense.
I kept it because the original transformation had it so I maintained it.
I think that this check is there because of the history of this transformation, as it is used by the code generator.
But honestly I have no idea.

What we could do is change the default of `ignore_strides` to `True` and then inside the code generator to `False` to maintain the old behaviour.

Reply:
My question is about checking that we are not copying from/to the same container in line 41. I have the impression that this wasn't there before but maybe I didn't check the diff properly.

Reply:
No you checked the diff correctly, I am just stupid.